design/apidsl/security.go

```diff
@@ -0,0 +1,308 @@
+		return
+	}
+
+	if securityMethodRedefined(name) {
+		return
+	}
+
+	def := &design.SecurityMethodDefinition{
+		Kind:   design.APIKeySecurityKind,
+		Method: name,
+		Type:   "apiKey",
+	}
+
+	if len(dsl) != 0 {
+		if !dslengine.Execute(dsl[0], def) {
+			return
+		}
+	}
+
+	design.Design.SecurityMethods = append(design.Design.SecurityMethods, def)
+}
+
+// OAuth2Security defines the different Security methods that are
+// available throughout the API.
+//
+// Example:
+//
+//     OAuth2Security("googAuth", func() {
+//	    	OAuth2Flow("accessCode")
+//	    	AuthorizationURL("http://example.com/authorization")
+//	    	TokenURL("http://example.com/token")
+//	    	Scope("my_system:write", "Write to the system")
+//	    	Scope("my_system:read", "Read anything in there")
+//     })
+//
+func OAuth2Security(name string, dsl ...func()) {
+	if _, ok := dslengine.CurrentDefinition().(*design.APIDefinition); !ok {
+		dslengine.IncompatibleDSL()
+		return
+	}
+
+	if securityMethodRedefined(name) {
+		return
+	}
+
+	def := &design.SecurityMethodDefinition{
+		Method: name,
+		Kind:   design.OAuth2SecurityKind,
+		Type:   "oauth2",
+	}
+
+	if len(dsl) != 0 {
+		if !dslengine.Execute(dsl[0], def) {
+			return
+		}
+	}
+
+	design.Design.SecurityMethods = append(design.Design.SecurityMethods, def)
+}
+
+// OtherSecurity defines a free-form security method, going above and
+// beyond the Swagger specs. If you want to benefit from Swagger UIs,
+// make sure the "securityType" is one of "apiKey", "oauth2" or
+// "basic".
+//
+// Example:
+//
+//     OtherSecurity("jwt", "apiKey", func() {
+//	    	InHeader("Authorization")
+//	    	AuthorizationURL("http://example.com/authorization")
+//	    	Scope("my_system:write", "Write to the system")
+//	    	Scope("my_system:read", "Read anything in there")
+//     })
+//
+func OtherSecurity(name string, securityType string, dsl ...func()) {
+	if _, ok := dslengine.CurrentDefinition().(*design.APIDefinition); !ok {
+		dslengine.IncompatibleDSL()
+		return
+	}
+
+	if securityMethodRedefined(name) {
+		return
+	}
+
+	def := &design.SecurityMethodDefinition{
+		Kind:   design.OtherSecurityKind,
+		Method: name,
+		Type:   securityType,
+	}
+
+	if len(dsl) != 0 {
+		if !dslengine.Execute(dsl[0], def) {
+			return
+		}
+	}
+
+	design.Design.SecurityMethods = append(design.Design.SecurityMethods, def)
+}
+
+// Scope defines an authorization scope. Used within SecurityMethod,
+// the description is required, explaining what the scope
+// means. Within a Security block, only a scope is needed.
+func Scope(name string, desc ...string) {
+	switch parent := dslengine.CurrentDefinition().(type) {
+	case *design.SecurityDefinition:
+		if len(desc) == 1 {
+			dslengine.ReportError("must not pass description to Scope()")
+			return
+		}
+		parent.Scopes = append(parent.Scopes, name)
+	case *design.SecurityMethodDefinition:
+		if len(desc) == 0 {
+			dslengine.ReportError("no description provided for Scope()")
```

Comment:
Well yes, in Swagger it becomes a `map[string]string` and it requires a description.

This invocation defines the available scopes throughout the API.. otherwise there's not much benefit to define it, other that to say what it is.